common/convert/convert.go

```diff
@@ -0,0 +1,195 @@
+package convert
+
+import (
+	"encoding/json"
+
+	"github.com/aserto-dev/ds-load/sdk/common/msg"
+	"github.com/aserto-dev/ds-load/sdk/transform"
+	dsc "github.com/aserto-dev/go-directory/aserto/directory/common/v3"
+	"github.com/aserto-dev/scim/common/model"
+	"github.com/elimity-com/scim"
+	"github.com/elimity-com/scim/optional"
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var (
+	ErrSourceUserTypeNotSet  = errors.New("source user type not set")
+	ErrSourceGroupTypeNotSet = errors.New("source group type not set")
+	ErrGroupsNotEnabled      = errors.New("groups not enabled")
+)
+
+type Converter struct {
```

Comment:
If it has no fields of its own, this could be defined as
```go
type Converter TransformConfig
```

Reply:
Actually, never mind. You don't want fields and methods of `TransformConfig` to be exposed from `Converter`.